cpp/src/tracing_dummy.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include <proton/tracing.hpp>
+#include <proton/annotation_key.hpp>
+#include <proton/message.hpp>
+
+namespace proton
+{
+void on_message_start_span(message &message, delivery &d) {}
+
+void on_message_end_span(delivery &d) {}
+
+void on_settled_span(delivery &d, tracker &track) {}
+
+void send_span(const message &message, binary &tag, tracker &track) {}
```

Comment:
These functions should be changed into an abstract class to make switching between the tracing and non tracing case easier.

Reply:
Done!